tests/Integration/RequestIntegrationTest.php

```diff
@@ -459,6 +462,25 @@ public function invokeDataProvider(): iterable
             null,
             null,
         ];
+
+        yield [
+            [
+                'max_request_body_size' => 'always',
+            ],
+            (new ServerRequest('POST', new Uri('http://www.example.com/foo')))
+                ->withHeader('Content-Type', 'application/json')
+                ->withBody($this->getStreamMock(0)),
+            [
+                'url' => 'http://www.example.com/foo',
+                'method' => 'POST',
+                'headers' => [
+                    'Host' => ['www.example.com'],
+                    'Content-Type' => ['application/json'],
+                ],
+            ],
+            null,
+            null,
+        ];
     }
 
     private function getStreamMock(?int $size, string $content = ''): StreamInterface
```

Comment:
Since the `$size` depends in the majority of the cases from the `$content` argument, WDYT of removing it, make `$content` nullable and compute `$size` accordingly?